src/migratus/migrations.clj

```diff
@@ -74,19 +92,49 @@
               (props/inject-properties properties content)
               content)))
 
-(defn find-migration-files [migration-dir exclude-scripts properties]
+(defn find-migration-files
+  "Looks for all migration files in 'migration-dir' path.
+   Excludes from results the migrations that match globs in 'exclude-scripts'
+
+   Parses the file names according to migratus rules.
+   Returns a sequence of maps.
+   Each map represents a single migration file.
+
+   A migration map has a single key - migration id as string.
+   The value is a map with migration name as key and
+   a another map as value representing the migration.
+
+   Example of structure:
+   { <migration-id>
+      { <migration-name>
+        {:sql {:up <contents> }    }}}
+   "
+  ;; ieugen: I wonder why we have to realize the migrations in memory
+  ;; The store could be enhanced to support a fetch-migration call to fetch the contents.
+  ;; Most of the time we are dealing with metdata.
+  ;; We need the migration body only when we apply it.
+  [migration-dir exclude-scripts properties]
   (log/debug "Looking for migrations in" migration-dir)
-  (->> (for [f (filter (fn [^File f] (.isFile f))
-                       (file-seq migration-dir))
-             :let [file-name (.getName ^File f)]
-             :when (not (utils/script-excluded? file-name migration-dir exclude-scripts))]
-         (if-let [mig (parse-name file-name)]
-           (migration-map mig (slurp f) properties)
-           (warn-on-invalid-migration file-name)))
-       (remove nil?)))
-
-
-(defn find-migration-resources [dir jar exclude-scripts properties]
+  (let [migration-dir (io/as-file migration-dir)]
+    (->> (for [f (filter (fn [^File f] (.isFile f))
+                         (file-seq migration-dir))
+               :let [file-name (.getName ^File f)]
+               :when (not (utils/script-excluded? file-name migration-dir exclude-scripts))]
+           (if-let [mig (parse-name file-name)]
+             (migration-map mig (slurp f) properties)
+             (warn-on-invalid-migration file-name)))
+         (remove nil?))))
+
+(comment
```

Comment:
test comment

Reply:
yeah, I was exploring. I probably will turn most of them in tests. best effort.